### PR TITLE
[PROD-782] BillingAccountChangeSynchronizer needs to update v2 workspaces using the billing project's old Google project

### PR DESCRIPTION
**Pull request by marctalbott**
Ticket: [PROD-782](https://broadworkbench.atlassian.net/browse/PROD-782)

---

**PR checklist**

- [ ] Include the JIRA issue number in the PR description and title
- [ ] Make sure Swagger is updated if API changes
  - [ ] **...and Orchestration's Swagger too!**
- [ ] If you changed anything in `model/`, then you should [publish a new official `rawls-model`](https://github.com/broadinstitute/rawls/blob/develop/README.md#publish-rawls-model) and update `rawls-model` in [Orchestration's dependencies](https://github.com/broadinstitute/firecloud-orchestration/blob/develop/project/Dependencies.scala).
- [ ] Get two thumbsworth of PR review
- [ ] Verify all tests go green, including CI tests
- [ ] **Squash commits and merge** to develop (branches are automatically deleted after merging)
- [ ] Inform other teams of any substantial changes via Slack and/or email

[PROD-782]: https://broadworkbench.atlassian.net/browse/PROD-782?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by cahrens — APPROVED** [review]

File: core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/BillingAccountChangeSynchronizer.scala [hunk 1]
```diff
@@ -204,7 +205,7 @@ final case class BillingAccountChangeSynchronizer(dataSource: SlickDataSource,
       v2Workspaces <- inTransaction {
         workspaceQuery
           .withBillingProject(billingProject.projectName)
-          .withoutGoogleProjectId(billingProject.googleProjectId)
+          .withVersion(WorkspaceVersions.V2)
```
Comment (ehigham) [inline core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/BillingAccountChangeSynchronizer.scala:208]: Ah my bad, this was a silly mistake. Hope it wasn't too difficult to spot.

**Review by ehigham — APPROVED** [review]

File: core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/BillingAccountChangeSynchronizer.scala (hunk 1 above)
Comment (cahrens) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/BillingAccountChangeSynchronizer.scala:208]: Awww, so nice of you to drop a note! We were trying not to bug you.

So there was no reason why you didn't check the version number? We were wondering if there was some transitory state where the version number might not be correct.

Comment (ehigham) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/BillingAccountChangeSynchronizer.scala:208]: Because I'm dumb

Comment (ehigham) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/BillingAccountChangeSynchronizer.scala:208]: And I didn't realise a workspace had a version number

Comment (cahrens) [reply core/src/main/scala/org/broadinstitute/dsde/rawls/monitor/BillingAccountChangeSynchronizer.scala:208]: Ah, well if you didn't know it had a version number I understand the comparison to the billing project name! Unfortunately though this caused an issue for migrated workspaces that kept their original billing project name.

**Review by aherbst-broad — APPROVED** [review]

**Comment by aherbst-broad** [comment]
jenkins retest

**Pull request merged**